### PR TITLE
Fix 6159 (Remove Unexpected Warning of PauliExpectation and AerPauliExpectation)

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

Resoleves #6159.

Previous (Aer)PauliExpectation converts a ListOp[PauliSumOp] into a ListOp[SummedOp[PauliOp]].
It raises the Warning.

### CODE REVIEW

**Review by woodsp-ibm — APPROVED** [review]

**Review by Cryoris — COMMENTED** [review]
Looks good to me, just one question: whats the reason for moving the ListOp check further up in the code?

**Comment by ikkoham** [comment]
@Cryoris Thanks. 

https://github.com/Qiskit/qiskit-terra/pull/6281/files#diff-b8a5aada04fd2fa0a8762d5ec05e29127f958549d8b47b6daa89f3edb6e7c5d5R48 
Sorry, this change has no reason. 

https://github.com/Qiskit/qiskit-terra/pull/6281/files#diff-b8a5aada04fd2fa0a8762d5ec05e29127f958549d8b47b6daa89f3edb6e7c5d5R69
This change is necessary. 

I was affected by the change below, so I changed the other parts as well, but it was not necessary in hindsight.
There is no difference between them, but it's an unnecessary change, so should I back?

File: qiskit/opflow/expectations/aer_pauli_expectation.py [hunk 1]
```diff
@@ -93,8 +94,6 @@ def _replace_pauli_sums(cls, operator):
             snapshot_instruction = SnapshotExpectationValue("expval_measurement", paulis)
             snapshot_op = CircuitStateFn(snapshot_instruction, is_measurement=True)
             return snapshot_op
-        if isinstance(operator, ListOp):
```
Comment (Cryoris) [inline qiskit/opflow/expectations/aer_pauli_expectation.py:96]: If we remove this, will it still work for objects that derive from list ops for which `operator.__class__ == ListOp` fails (like e.g. `ComposedOp`)?

**Comment by Cryoris** [comment]
> I was affected by the change below, so I changed the other parts as well, but it was not necessary in hindsight.
There is no difference between them, but it's an unnecessary change, so should I back?

I think it's good for bugfixes to change only what's necessary, that makes it easier to review 😉 In this case I'm fine because you said it's not part of the bugfix 👍🏻

File: qiskit/opflow/expectations/aer_pauli_expectation.py (hunk 1 above)
Comment (ikkoham) [reply qiskit/opflow/expectations/aer_pauli_expectation.py:96]: that case is not supported originally. 
```python
converted = AerPauliExpectation().convert(StateFn(ComposedOp([X, Z]), is_measurement=True) @ Zero)
CircuitSampler(Aer.get_backend("qasm_simulator")).convert(converted)
```
main
```
ValueError: Composition with a Statefunction in the first operand is not defined.
```
this fix
```
TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
```

In the original case, the conversion can be done, but the conversion result is not correct (undefined behavior).
I think that an error should be raised in early stage.

**Comment by ikkoham** [comment]
Could you add a label `Changelog: Bugfix`?

**Review by woodsp-ibm — APPROVED** [review]

**Review by Cryoris — APPROVED** [review]

**Pull request merged**